api/client.go

```diff
@@ -329,13 +331,52 @@ func (c *Client) AddLocalCharm(curl *charm.URL, ch charm.Charm) (*charm.URL, err
 		return nil, errors.Errorf("unknown charm type %T", ch)
 	}
 
-	curl, err := c.UploadCharm(curl, archive)
+	anyHooks, err := hasHooks(archive.Name())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if !anyHooks {
+		return nil, errors.Errorf("invalid charm %q: has no hooks", curl.Name)
```

Comment:
perhaps suggest to run charm build if it's a reactive charm?

Reply:
We do not really make suggestions in errors coming from apiserver. As discussed elsewhere, current message is sufficient.